src/core/scripts.js

```diff
@@ -3,14 +3,165 @@ import fs from 'fs/promises';
 import { watch } from 'fs';
 import path from 'path';
 
+const components = ['x', 'y', 'z', 'w'];
+
+function rawToValue(app, args, value, old) {
+    const vecLookup = [undefined, undefined, global.pc.Vec2, global.pc.Vec3, global.pc.Vec4];
+
+    switch (args.type) {
+        case 'boolean':
+            return !!value;
+
+        case 'number':
+            if (typeof value === 'number') {
+                return value;
+            } else if (typeof value === 'string') {
+                var v = parseInt(value, 10);
+                if (isNaN(v)) return null;
+                return v;
+            } else if (typeof value === 'boolean') {
+                return 0 + value;
+            }
+
+            return null;
+
+        case 'json':
+            {
```

Comment:
Do not copy code from engine directly, please rewrite and ensure you get this code well.